ompi/mca/coll/ucc/coll_ucc_module.c

```diff
@@ -602,6 +671,52 @@ int mca_coll_ucc_req_free(struct ompi_request_t **ompi_req)
 void mca_coll_ucc_completion(void *data, ucc_status_t status)
 {
     mca_coll_ucc_req_t *coll_req = (mca_coll_ucc_req_t*)data;
-    ucc_collective_finalize(coll_req->ucc_req);
+    if (false == coll_req->super.req_persistent) {
+        ucc_collective_finalize(coll_req->ucc_req);
+    } else {
+        UCC_VERBOSE(5, "%s done %p", "<coll>_init", coll_req);
+        assert(!REQUEST_COMPLETE(&coll_req->super));
+    }
     ompi_request_complete(&coll_req->super, true);
 }
+
+/* req_start() : ompi_request_start_fn_t */
+int mca_coll_ucc_req_start(size_t count, struct ompi_request_t **requests)
+{
+    size_t ii;
+    int rc = OMPI_SUCCESS;
+
+    for (ii = 0; ii < count; ++ii) {
+        mca_coll_ucc_req_t *coll_req = (mca_coll_ucc_req_t *) requests[ii];
+        ucc_status_t rc_ucc;
+
+        if ((NULL == coll_req) || (OMPI_REQUEST_COLL != coll_req->super.req_type)) {
+            continue;
+        }
+        if (true != coll_req->super.req_persistent) {
+            continue;
+        }
+        UCC_VERBOSE(5, "%s post %p", "<coll>_init", coll_req);
+        assert(REQUEST_COMPLETE(&coll_req->super));
+        assert(OMPI_REQUEST_INACTIVE == coll_req->super.req_state);
+
+        coll_req->super.req_status.MPI_TAG = MPI_ANY_TAG;
+        coll_req->super.req_status.MPI_ERROR = OMPI_SUCCESS;
+        coll_req->super.req_status._cancelled = 0;
+        coll_req->super.req_complete = REQUEST_PENDING;
+        coll_req->super.req_state = OMPI_REQUEST_ACTIVE;
+
+        rc_ucc = ucc_collective_post(coll_req->ucc_req);
+        if (UCC_OK != rc_ucc) {
+            UCC_ERROR("ucc_collective_post failed: %s", ucc_status_string(rc_ucc));
+            coll_req->super.req_complete = REQUEST_COMPLETED;
+            coll_req->super.req_state = OMPI_REQUEST_INACTIVE;
+            if (OMPI_SUCCESS == rc) {
```

Comment:
please ensure that `coll_req->super.req_status.MPI_ERROR` reflect the `rc` error.